exonum/src/runtime/types.rs

```diff
@@ -63,12 +91,39 @@ impl AnyTx {
     }
 }
 
+/// The artifact identifier is required by the runtime to construct service instances.
+/// In other words an artifact identifier means same as class name, and a specific service
+/// instance is the class instance.
+///
+/// In string representation, the artifact identifier is written as follows:
+///
+/// `{runtime_id}:{artifact_name}`, where `runtime_id` is well known [runtime identifier],
+/// and `artifact_name` is unique name of artifact.
+///
+/// Artifact name can contains only these characters: `a-zA-Z0-9` and one of `_-./`.
+///
+/// [runtime identifier]: enum.RuntimeIdentifier.html
+///
+/// # Example
+///
+/// ```
+/// # use exonum::runtime::ArtifactId;
+/// # fn main() -> Result<(), failure::Error> {
+/// // Typical Rust artifact.
+/// let rust_artifact_id = "0:my-service/1.0.0".parse::<ArtifactId>()?;
+/// // Typical Java artifact.
+/// let java_artifact_id = "1:org.exonum.service.1".parse::<ArtifactId>()?;
+/// # Ok(())
+/// # }
+/// ```
 #[derive(
     Debug, Clone, ProtobufConvert, Serialize, Deserialize, PartialEq, Eq, Hash, PartialOrd, Ord,
 )]
 #[exonum(pb = "schema::runtime::ArtifactId", crate = "crate")]
 pub struct ArtifactId {
+    /// Well known runtime identifier.
```

Comment:
`well-known`. Though I can't understand what does it mean here

Reply:
This means that the runtimes identifiers are the same in different blockchains.

Reply:
Probably you mean in _different runtimes_ ?

Reply:
Indeed different Exonum networks should have same runtime identifiers for the same runtimes in my opinion.

Reply:
I'd call it `predefined` and clarify the fact that it's the same across all Exonum-based blockchains